worker/modelcache/worker.go

```diff
@@ -181,6 +191,22 @@ func (c *cacheWorker) loop() error {
 					}
 				}
 			}
+
+			// In theory we could just create a new controller and just apply
+			// the new delta's to that. We can track if a new controller is
+			// required in the same vein as sweepRequired.
+			c.mu.Lock()
```

Comment:
What are we locking for? The only other place where sweepRequired it touched is in the same goroutine.